launcher/net/Upload.cpp

```diff
@@ -219,7 +219,12 @@ namespace Net {
         if (APPLICATION->capabilities() & Application::SupportsFlame
                 && request.url().host().contains("api.curseforge.com")) {
             request.setRawHeader("x-api-key", APPLICATION->getFlameAPIKey().toUtf8());
+        } else if (request.url().host().contains("api.modrinth.com")) {
```

Comment:
this check should probably use [`BuildConfig::MODRINTH_PROD_URL`](https://github.com/PrismLauncher/PrismLauncher/blob/9f202caf9356ebaf5fd845210adea5e935d4350f/buildconfig/BuildConfig.h#L162) instead (well, same thing for the CF thing above this tbh).

Reply:
Hmm, I feel like this is better since it only checks the host. Although contains isn't great (why not ==).

Reply:
The issue is if the host changes in the future. For now it's very unlikely that they will, but the purpose of having stuff in `BuildConfig` is to prevent problems with that in the future. A simple `startsWith` here would suffice, or perhaps make the one in `BuildConfig` a `QUrl` and compare their `host()` with `==` could work too.

Reply:
IMO it should be an exact match to avoid leaking the token to anyone.

A simple example could be a modpack that downloads something from a domain like `api.modrinth.com.scrumplex.net`

Reply:
If you did startsWith, you'd need to append a /, so a think making it a QUrl is smart.

Reply:
Soo... which approach should I use?
- the current with == instead
- replace all the constants with QUrl
- only make modrinth and curseforge QUrls
- do startsWith(BuildConfig::THING + '/')

Reply:
I think it should be an exact match (`==`). I guess changing the URLs in BuildConfig would be helpful for this.

Reply:
I think this would cause too many other changes so I've added a TODO.